### DISCUSSION TITLE
Adding values to a setup store in a unit test

### DISCUSSION THREAD

**Discussion by BrendonLapp** [vuejs/pinia#1610] [Help and Questions] [answered]
Hi, I'm looking to add values to a unit tested pinia store when I'm testing my component. Is there an easy way to do this?

My store looks like this:

```
export const useStore = defineStore("Store", () => {
	const store = reactive({
		messages: [],
	});
	
	return { store };
});
```

And in my tests I am trying to do something like this:

```
	const wrapper = mount(Component, {
		global: {
			plugins: [createTestingPinia()],
		},
	});

//..... in my test
        const store = useStore();
	store.store = {
			messages: [
				{
					message: "First",
					color: Color.danger,
					timeout: 10000,
				},
			] ,
		};
```

What am I missing to be able to get this value to be set in my store?

Right now, my `timeout` always ends up being undefined. But what I look at what the store actually is being set as I end up with the whole store being empty each time. I feel like I'm missing something really simple to set this value

**Comment by posva** [comment, marked as answer]
It should be a ref instead if you want to be able to change it. You could directly use

```js
Const messages = ref([])
```

**Reply by BrendonLapp** [reply]
Is ref the better option over using reactive?

I did find a way to do what I needed by doing this:
```js
const wrapper = mount(ToastMessage, {
			props: {
				message: "Success Message",
				color: ToastColor.success,
				index: 0,
			},
			global: {
				plugins: [
					createTestingPinia({
						initialState: {
							toastStore: {
								store: {
									messages: [
										{
											message: "Success Message",
											color: ToastColor.success,
											timeout: 10000,
										},
									],
								},
							},
						},
					}),
				],
			},
		});
```
And it seems to be working decently for me

**Reply by posva** [reply]
in this case yes because a reactive cannot be replaced
